From d62daa15cb01e4cce2a00ae94bf22263dbb3f86c Mon Sep 17 00:00:00 2001
From: Yoann Lopes <yoann.lopes@digia.com>
Date: Thu, 3 Apr 2014 15:17:49 +0200
Subject: [PATCH] GStreamer: cleanup system includes.

Replaced some linux includes with <private/qcore_unix_p.h> to be more
portable.
Use qt_safe_open() and qt_safe_close() instead of ::open() and
::close().

Task-number: QTBUG-38061
Change-Id: Idb3f50d070ec1701cafba98e79e2e85824653a3a
---
 src/gsttools/gsttools.pro                                |  2 +-
 src/gsttools/qgstreamervideoinputdevicecontrol.cpp       | 15 +++------------
 .../audiodecoder/qgstreameraudiodecoderserviceplugin.cpp | 13 -------------
 src/plugins/gstreamer/camerabin/camerabincontrol.cpp     | 12 ------------
 .../gstreamer/camerabin/camerabinserviceplugin.cpp       | 16 +++-------------
 src/plugins/gstreamer/common.pri                         |  2 +-
 .../mediacapture/qgstreamercaptureserviceplugin.cpp      | 16 +++-------------
 .../gstreamer/mediacapture/qgstreamerv4l2input.cpp       | 11 +----------
 .../mediaplayer/qgstreamerplayerserviceplugin.cpp        | 14 --------------
 9 files changed, 12 insertions(+), 89 deletions(-)

--- a/src/gsttools/gsttools.pro
+++ b/src/gsttools/gsttools.pro
@@ -2,7 +2,7 @@ TEMPLATE = lib
 
 TARGET = qgsttools_p
 QPRO_PWD = $$PWD
-QT = core multimedia-private gui-private
+QT = core-private multimedia-private gui-private
 
 !static:DEFINES += QT_MAKEDLL
 
--- a/src/gsttools/qgstreamervideoinputdevicecontrol.cpp
+++ b/src/gsttools/qgstreamervideoinputdevicecontrol.cpp
@@ -44,16 +44,7 @@
 #include <QtCore/QDir>
 #include <QtCore/QDebug>
 
-#include <linux/types.h>
-#include <sys/time.h>
-#include <sys/ioctl.h>
-#include <sys/poll.h>
-#include <unistd.h>
-#include <fcntl.h>
-#include <errno.h>
-#include <string.h>
-#include <stdlib.h>
-#include <sys/mman.h>
+#include <private/qcore_unix_p.h>
 #include <linux/videodev2.h>
 
 QGstreamerVideoInputDeviceControl::QGstreamerVideoInputDeviceControl(QObject *parent)
@@ -135,7 +126,7 @@ void QGstreamerVideoInputDeviceControl::
     foreach( const QFileInfo &entryInfo, entries ) {
         //qDebug() << "Try" << entryInfo.filePath();
 
-        int fd = ::open(entryInfo.filePath().toLatin1().constData(), O_RDWR );
+        int fd = qt_safe_open(entryInfo.filePath().toLatin1().constData(), O_RDWR );
         if (fd == -1)
             continue;
 
@@ -165,6 +156,6 @@ void QGstreamerVideoInputDeviceControl::
             m_names.append(entryInfo.filePath());
             m_descriptions.append(name);
         }
-        ::close(fd);
+        qt_safe_close(fd);
     }
 }
--- a/src/plugins/gstreamer/audiodecoder/qgstreameraudiodecoderserviceplugin.cpp
+++ b/src/plugins/gstreamer/audiodecoder/qgstreameraudiodecoderserviceplugin.cpp
@@ -49,19 +49,6 @@
 #include <QtCore/QDir>
 #include <QtCore/QDebug>
 
-#include <linux/types.h>
-#include <sys/time.h>
-#include <sys/ioctl.h>
-#include <sys/poll.h>
-#include <unistd.h>
-#include <fcntl.h>
-#include <errno.h>
-#include <string.h>
-#include <stdlib.h>
-#include <sys/mman.h>
-#include <linux/videodev2.h>
-#include <gst/gst.h>
-
 // #define QT_SUPPORTEDMIMETYPES_DEBUG
 
 QMediaService* QGstreamerAudioDecoderServicePlugin::create(const QString &key)
--- a/src/plugins/gstreamer/camerabin/camerabincontrol.cpp
+++ b/src/plugins/gstreamer/camerabin/camerabincontrol.cpp
@@ -50,18 +50,6 @@
 #include <QtCore/qfile.h>
 #include <QtCore/qmetaobject.h>
 
-#include <linux/types.h>
-#include <sys/time.h>
-#include <sys/ioctl.h>
-#include <sys/poll.h>
-#include <unistd.h>
-#include <fcntl.h>
-#include <errno.h>
-#include <string.h>
-#include <stdlib.h>
-#include <sys/mman.h>
-#include <linux/videodev2.h>
-
 QT_BEGIN_NAMESPACE
 
 //#define CAMEABIN_DEBUG 1
--- a/src/plugins/gstreamer/camerabin/camerabinserviceplugin.cpp
+++ b/src/plugins/gstreamer/camerabin/camerabinserviceplugin.cpp
@@ -50,18 +50,8 @@
 #include "camerabinservice.h"
 #include <private/qgstutils_p.h>
 
-#include <linux/types.h>
-#include <sys/time.h>
-#include <sys/ioctl.h>
-#include <sys/poll.h>
-#include <unistd.h>
-#include <fcntl.h>
-#include <errno.h>
-#include <string.h>
-#include <stdlib.h>
-#include <sys/mman.h>
+#include <private/qcore_unix_p.h>
 #include <linux/videodev2.h>
-#include <gst/gst.h>
 
 QT_BEGIN_NAMESPACE
 
@@ -135,7 +125,7 @@ void CameraBinServicePlugin::updateDevic
     QFileInfoList entries = devDir.entryInfoList(QStringList() << "video*");
 
     foreach (const QFileInfo &entryInfo, entries) {
-        int fd = ::open(entryInfo.filePath().toLatin1().constData(), O_RDWR );
+        int fd = qt_safe_open(entryInfo.filePath().toLatin1().constData(), O_RDWR );
         if (fd == -1)
             continue;
 
@@ -165,7 +155,7 @@ void CameraBinServicePlugin::updateDevic
             m_cameraDevices.append(entryInfo.filePath().toLocal8Bit());
             m_cameraDescriptions.append(name);
         }
-        ::close(fd);
+        qt_safe_close(fd);
     }
 }
 
--- a/src/plugins/gstreamer/common.pri
+++ b/src/plugins/gstreamer/common.pri
@@ -1,5 +1,5 @@
 
-QT += multimedia-private network
+QT += core-private multimedia-private network
 CONFIG += no_private_qt_headers_warning
 
 qtHaveModule(widgets) {
--- a/src/plugins/gstreamer/mediacapture/qgstreamercaptureserviceplugin.cpp
+++ b/src/plugins/gstreamer/mediacapture/qgstreamercaptureserviceplugin.cpp
@@ -51,18 +51,8 @@
 #include "qgstreamercaptureservice.h"
 #include <private/qgstutils_p.h>
 
-#include <linux/types.h>
-#include <sys/time.h>
-#include <sys/ioctl.h>
-#include <sys/poll.h>
-#include <unistd.h>
-#include <fcntl.h>
-#include <errno.h>
-#include <string.h>
-#include <stdlib.h>
-#include <sys/mman.h>
+#include <private/qcore_unix_p.h>
 #include <linux/videodev2.h>
-#include <gst/gst.h>
 
 QMediaService* QGstreamerCaptureServicePlugin::create(const QString &key)
 {
@@ -142,7 +132,7 @@ void QGstreamerCaptureServicePlugin::upd
     foreach( const QFileInfo &entryInfo, entries ) {
         //qDebug() << "Try" << entryInfo.filePath();
 
-        int fd = ::open(entryInfo.filePath().toLatin1().constData(), O_RDWR );
+        int fd = qt_safe_open(entryInfo.filePath().toLatin1().constData(), O_RDWR );
         if (fd == -1)
             continue;
 
@@ -172,7 +162,7 @@ void QGstreamerCaptureServicePlugin::upd
             m_cameraDevices.append(entryInfo.filePath().toLocal8Bit());
             m_cameraDescriptions.append(name);
         }
-        ::close(fd);
+        qt_safe_close(fd);
     }
 }
 #endif
--- a/src/plugins/gstreamer/mediacapture/qgstreamerv4l2input.cpp
+++ b/src/plugins/gstreamer/mediacapture/qgstreamerv4l2input.cpp
@@ -44,16 +44,7 @@
 #include <QtCore/qdebug.h>
 #include <QtCore/qfile.h>
 
-#include <linux/types.h>
-#include <sys/time.h>
-#include <sys/ioctl.h>
-#include <sys/poll.h>
-#include <unistd.h>
-#include <fcntl.h>
-#include <errno.h>
-#include <string.h>
-#include <stdlib.h>
-#include <sys/mman.h>
+#include <private/qcore_unix_p.h>
 #include <linux/videodev2.h>
 
 QT_BEGIN_NAMESPACE
--- a/src/plugins/gstreamer/mediaplayer/qgstreamerplayerserviceplugin.cpp
+++ b/src/plugins/gstreamer/mediaplayer/qgstreamerplayerserviceplugin.cpp
@@ -51,20 +51,6 @@
 #include "qgstreamerplayerservice.h"
 #include <private/qgstutils_p.h>
 
-#include <linux/types.h>
-#include <sys/time.h>
-#include <sys/ioctl.h>
-#include <sys/poll.h>
-#include <unistd.h>
-#include <fcntl.h>
-#include <errno.h>
-#include <string.h>
-#include <stdlib.h>
-#include <sys/mman.h>
-#include <linux/videodev2.h>
-#include <gst/gst.h>
-
-
 QMediaService* QGstreamerPlayerServicePlugin::create(const QString &key)
 {
     QGstUtils::initializeGst();
